src/inventory_ui.cpp

```diff
@@ -276,24 +277,58 @@ void inventory_selector::print_inv_weight_vol(int weight_carried, int vol_carrie
         int vol_capacity) const
 {
     // Print weight
-    mvwprintw(w_inv, 0, 32, _("Weight (%s): "), weight_units());
-    nc_color weight_color;
+    mvwprintw(w_inv, 0, TERMX - 41, _("Weight: "), weight_units());
+    nc_color weight_color = c_ltgray;
     if (weight_carried > u.weight_capacity()) {
         weight_color = c_red;
-    } else {
-        weight_color = c_ltgray;
     }
     wprintz(w_inv, weight_color, "%6.1f", convert_weight(weight_carried) + 0.05 ); // +0.05 to round up;
-    wprintz(w_inv, c_ltgray, "/%-6.1f", convert_weight(u.weight_capacity()));
+    wprintz(w_inv, c_ltgray, "/%-6.1f %s", convert_weight(u.weight_capacity()), weight_units());
+    wmove(w_inv, 0, TERMX - 15);
+    wprintz(w_inv, c_ltgray, "[............]");
+    print_bar(0, TERMX - 14, weight_carried, g->u.weight_capacity());
 
     // Print volume
-    mvwprintw(w_inv, 0, 61, _("Volume: "));
+    mvwprintw(w_inv, 1, TERMX - 41, _("Volume: "));
+    wmove(w_inv, 1, TERMX - 30);
     if (vol_carried > vol_capacity) {
         wprintz(w_inv, c_red, "%3d", vol_carried);
     } else {
         wprintz(w_inv, c_ltgray, "%3d", vol_carried);
     }
     wprintw(w_inv, "/%-3d", vol_capacity);
+    wmove(w_inv, 1, TERMX - 15);
+    wprintz(w_inv, c_ltgray, "[............]");
+    print_bar(1, TERMX - 14, vol_carried, vol_capacity);
+}
+
+void inventory_selector::print_bar(int y, int x, int carried, int capacity) const {
```

Comment:
I can see where it accepts custom characters (use one character for part of bar, use another for next part, etc) however I'm missing the part where `get_labeled_bar` will color the characters.

Perhaps you're thinking I'd like to color the entire bar one color, including the two end-cap brackets?  I was hoping to color the brackets separately.  And from the looks of things `get_labeled_bar` only returns a `std::string`... perhaps in that case, print out what I get from the function in color, then overwrite the brackets in `c_ltgray`.

Nevermind, I'll leave the thing untouched.